wikipedia_scraper/spiders/wikiscrape.py

```diff
@@ -0,0 +1,87 @@
+import os
+import scrapy
+import logging
+import argparse
+from datetime import datetime
+from scrapy.crawler import CrawlerProcess
+from scrapy.utils.project import get_project_settings
+
+log = logging.getLogger(__name__)
+
+class WikiSpider(scrapy.Spider):
+    def __init__(self, file=None):
+        if file:
+            with file as f:
+                self.start_urls = [url.strip() for url in f.readlines()]
+        else:
+            self.start_urls = ['https://en.wikipedia.org/wiki/Dog']        
+            
+    name = "wiki"
+    allowed_domains = ['wikipedia.org']
+
+    def parse(self, response):
+        log.info('Parse function called on %s on spider %s', response.url,self.name)
+
+        last_modified_date = None
+        if response.headers.has_key('last-modified'):
+            last_modified_date_raw = response.headers["last-modified"].decode('ascii')
+            last_modified_date = datetime.fromtimestamp(
+                datetime.strptime(last_modified_date_raw, "%a, %d %b %Y %H:%M:%S GMT").timestamp())
+
+        yield {
+            'url': response.url,
+            'title': response.css('title::text').get(),
+            'lastMod': last_modified_date,
+            'TOC': response.css('.toc .toctext::text').getall(),
+            'mainText': ' '.join(response.css('#content p::text, #content a::text').re('[A-Za-z0-9\-]+')).lower(),
+            'boldText': ' '.join(response.css('#content p b::text').re('.{2,}')).lower(),
+            'italicText': ' '.join(response.css('#content p i::text').re('.{2,}')).lower(),
+            'img': response.css('.infobox img::attr(src), .infobox-full-data img::attr(src), .thumbinner img::attr(src)').get()
+        }
+
+        hrefLinks = set(response.css('#content a::attr(href)').re('^\/wiki\/(?!File:).*$'))
+        yield from response.follow_all(hrefLinks, callback=self.parse)
+
+def main():
+    parser = argparse.ArgumentParser(description="Crawler options. To configure mongoDB connection, please edit project settings.py")
```

Comment:
We shouldn't introduce an extra main function. Instead we should use scrapy's built in method using the cli to run the project. Doing this would allow us to use scrapy's cli to change config settings without cluttering the project without needing to add a new library and a custom cli